src/utils/index_typing.coffee

```diff
@@ -0,0 +1,96 @@
+
+# Get the index rule for a cozydb type
+#
+# Returns an index rule with
+#    :indexType {String} - the data-indexer type for this field
+#    :indexTransform {Function} - the transform to apply to the value before
+getIndexType = (type) ->
+
+    # Support for both styles
+    #   field: String
+    #   field: SubModel
+    #   field: {type: String, indexTransform: parseInt, indexType: 'number'}
+    if type.type and not type.indexType
+        type = type.type
+
+    if type.indexTransform and type.indexType
+        indexType: type.indexType
+        indexTransform: type.indexTransform
+
+    else if type is String
+        indexType: 'string'
+        indexTransform: null
+
+    else if type is Date
+        indexType: 'date'
+        indexTransform: null
+
+    else if type is Number
+        indexType: 'number'
+        indexTransform: null
+
+    else if type is Boolean
+        indexType: 'boolean'
+        indexTransform: null
+
+    # case where the type is an array like [String], [SubModel]
+    else if Array.isArray(type) and type.length is 1
+
+        itemIndexRule = getIndexType(type[0])
+        if itemIndexRule?.indexType is 'string'
+
+            indexTransform = if itemIndexRule.indexTransform
+            then (data) -> data.map(itemIndexRule.indexTransform).join ' '
+            else (data) -> data.join ' '
+
+            indexType: 'string'
+            indexTransform: indexTransform
+
+
+        else
+            # for now, we only support array of string or toStringable
+            undefined
+    else
+        undefined
```

Comment:
@aenario you should probably warn if the type is not recognized. It will avoid stupid mistakes, like `bool` instead of `boolean`